catkin_tools/jobs/utils.py

```diff
@@ -50,42 +50,67 @@ def get_env_loaders(package, context):
     return sources
 
 
+def merge_envs(job_env, overlay_envs):
+    '''
+    In the merged/linked case of single env, this function amounts to a straight
+    assignment, but a more complex merge is required with isolated result spaces,
+    since a package's build environment may require extending that of multiple
+    other result spaces.
+    '''
+    merge_path_values = {}
+
+    for overlay_env in overlay_envs:
+        for key, values_str in overlay_env.items():
+            if key.endswith('PATH'):
```

Comment:
It isn't necessary in the non-isolated case; it was only made always on because I was thinking it would be non-harmful, and nice to keep a lesser-used piece of code in service to prevent it from rotting. I'm adding a commit which avoids the merge when there's only a single new env.